### PR TITLE
Explicit types for post and put request bodies

### PR DESCRIPTION
**Pull request by kanishka-azimi**
These are the types I want Servant to parse into. Can switch from plain types to records at a later point. I have wanted to perform this small step for months.

### CODE REVIEW

**Comment by kanishka-azimi** [comment]
Eventually, there should be at least one generator in "model.factories" to loosely correspondence with each request type here.

**Comment by kanishka-azimi** [comment]
@chreekat I am merging this. It's a superficial change and seems universal good and the PR has been open for a few days.

I am happy to rollback this change or modify it, if you do have feedback later.

File: src/Controller/Asset.hs [hunk 1]
```diff
@@ -172,6 +172,15 @@ detectUpload u = do
...
     =<< lift (probeFile (fileUploadName u) =<< peeks (fileUploadPath u))
 
+data ProcessAssetRequest =
+    ProcessAssetRequest
+        (Maybe (FileInfo TempFile))
+        (Maybe (Id Token))
+        (Maybe T.Text)
+        (Maybe Release)
+        (Maybe (Id Container))
+        ()
```
Comment (chreekat) [inline src/Controller/Asset.hs]: To verify, you're saying you'd like `processAsset` to have type `:: AssetTarget -> ProcessAssetRequest -> handler response` ?

**Comment by kanishka-azimi** [comment]
> My only concern, as usual, is that now we have a bunch of "dead" code to keep track of. However, making an account of our API types is a huge win. Let's just do servant sooner rather than later!

Another two hours and we could probably delete the function that I partially replaced.

**Comment by kanishka-azimi** [comment]
Actually, wrong PR, ignore last comment.

File: src/Controller/Asset.hs (hunk 1 above)
Comment (kanishka-azimi) [reply src/Controller/Asset.hs]: Haven't decided yet, but that sounds promising. processAsset itself is way too big.

An alternative would be to extend AssetTarget to include more data.

**Pull request merged**